iModelCore/Bentley/PublicAPI/Bentley/WString.h

```diff
@@ -542,4 +542,19 @@ typedef T_Utf8StringVector*         T_Utf8StringVectorP, &T_Utf8StringVectorR;
 typedef T_Utf8StringVector const*   T_Utf8StringVectorCP;
 typedef T_Utf8StringVector const&   T_Utf8StringVectorCR;
 
+// Supports char8_t (including u8 literals) interop with Utf8String/Utf8CP APIs.
+template <size_t N>
+class Utf8Chars {
+    Utf8Char m_characters[N];
+    
+    template<size_t... I>
+    constexpr Utf8Chars(const char8_t (&input)[N], std::index_sequence<I...>) : m_characters {static_cast<Utf8Char>(input[I])...} {}
+    
+public:
+    constexpr Utf8Chars(const char8_t (&input)[N]) : Utf8Chars(input, std::make_index_sequence<N>()) {}
+
+    constexpr operator Utf8CP() const { return m_characters; }
+    constexpr Utf8CP operator&() const { return m_characters; }

```

Comment:
Yuck!

Reply:
I agree. Supposedly, wonderful new utf8 APIs are coming in future C++ versions. However, for now, we are stuck doing this. They made char8_t a completely separate type in C++20...